File: 루시드/20221011_2.py
def solution(rows, columns, queries):
    answer = []
    board = [[i + j * columns for i in range(1, columns + 1)] for j in range(rows)]

    for r1, c1, r2, c2 in queries:
        r1, c1, r2, c2 = r1 - 1, c1 - 1, r2 - 1, c2 - 1
        stack = []

        # 오른쪽
        for i in range(c1, c2 + 1):
            stack.append(board[r1][i])
            if len(stack) == 1:
                continue
            board[r1][i] = stack[-2]

        # 아래
        for j in range(r1 + 1, r2 + 1):
            stack.append(board[j][i])
            board[j][i] = stack[-2]

        # 왼쪽
        for k in range(c2 - 1, c1 - 1, -1):
            stack.append(board[j][k])
            board[j][k] = stack[-2]

        # 위쪽
        for l in range(r2 - 1, r1 - 1, -1):
            stack.append(board[l][k])
            board[l][k] = stack[-2]

        answer.append(min(stack))

    return answer

File: 아더/20221011_2.java
package 아더;

import java.util.Arrays;

public class PRO_77485_행렬테두리회전하기 {

    public static void main(String[] args) {

        PRO_77485_행렬테두리회전하기 pro = new PRO_77485_행렬테두리회전하기();

        int[] rows = {6, 3, 100};
        int[] columns = {6, 3, 97};
        int[][][] queries = {
            {{2,2,5,4},{3,3,6,6},{5,1,6,3}},
            {{1,1,2,2},{1,2,2,3},{2,1,3,2},{2,2,3,3}},
            {{1,1,100,97}}
        };

        for (int i = 0; i < 3; i++) {
            System.out.println(Arrays.toString(pro.solution(rows[i], columns[i], queries[i])));
        }

    }

    public int[] solution(int rows, int columns, int[][] queries) {
        int[] answer = {};



        return answer;
    }

}
